src/UI/class-settings-page.php

```diff
@@ -598,6 +612,30 @@ public function print_multiple_checkboxes( array $args ): void {
 		$this->print_description_text( $args );
 	}
 
+	/**
+	 * Print out a "single-image browse control" which includes a text
+	 * input to store image path and a button to browse for images.
+	 *
+	 * @param array $args The arguments for the control.
+	 * @return void
+	 */
+	public function print_media_single_image( array $args ) {
+		$key         = esc_attr( $args['option_key'] );
```

Comment:
The output escaping is already included in the markup below - it's not needed here as well.

Reply:
We currently do this consistently in many other places in the code (for example see `$id` variable in `print_select_tag()`, `print_binary_radio_tag()`, `print_text_tag()` etc.

I assume that removing this doesn't have any security risk since we escape when outputting it, but if we are to do it then it would be better to do it everywhere to keep consistency. Else it just looks like an omission.

Would you still go for removing this?

Reply:
Yes - escaping twice is unnecessary.

Escaping should be done as late as possible (i.e. when pushing into echoed markup). Until we know the _context_ with which a value be displayed (HTML vs attribute vs XML, etc.) then we can't confidently escape it, so this escaping on line 623 is incorrect.

Reply:
Pretty good point. Removed.